File: C/Euclid_GCD.c
// Program to find GCD of two numbers using Euclidean GCD algorithm
// This program returns GCD in O(log(N)) time. 

#include <stdio.h>

int euclids_gcd(int a,int b) // recursive function 
{
    if(b == 0)
    {
        return a;
    }
    return euclids_gcd(b, a%b); 
}

int print_gcd(int a,int b)
{
    int ans = euclids_gcd(a,b);
    printf("GCD of above numbers is %d",ans);
}
int main()
{
    int a,b;
    printf("Enter 2 number's for which GCD is required \n");
    scanf("%d %d",&a,&b);
    print_gcd(a,b); 
 
    return 0;
}

/*

Enter 2 number's for which GCD is required                                                                                      
4 8                                                                                                                             
GCD of above numbers is 4       

*/

File: PHP/euclid_gcd.php
<?php # euclid_gcd.php - Nate Nasteff

/*
** The Euclidian Algorithm for getting the greatest common divisor of two
** integers operates by performing modulo operation (b % a) recursivley (which
** then turns to a % b) until the the result of the modulo operation reaches 0.
** This function takes two int args and returns an int (GCD)
*/

$get_gcd = function ($int_a, $int_b) use (&$get_gcd) {
  return ($int_a == 0) ? $int_b : $get_gcd($int_b % $int_a, $int_a);
};

// DataStructures.Test Case

$int_a = 325;
$int_b = 5;

echo "\nYour numbers:\n" . $int_a . " " . $int_b
  . "\n\nGreatest common divisor:\n" . $get_gcd($int_a, $int_b)
  . "\n";

/*
** Sample I/O
**
** INPUT:
** 
** 
**
** OUTPUT:
**
** Greatest common divisor of 111 and 9: 3
**
**
** Time complexity - O(log(a + b))
**
** Space complexity - O(1)
**
 */